src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/ConsensusRuleDescriptorTest.cs

```diff
@@ -8,70 +10,56 @@ namespace Stratis.Bitcoin.Features.Consensus.Tests.Rules
 
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            descriptor.Rule.Should().BeOfType<RuleWithValidationExecutionMempoolAttributes>();
+            descriptor.RuleAttributes.Should().HaveCount(3);
+            descriptor.RuleAttributes.Should().Contain(x => x is MempoolRuleAttribute);
+            descriptor.RuleAttributes.Should().Contain(x => x is ExecutionRuleAttribute);
+            descriptor.RuleAttributes.Should().Contain(x => x is ValidationRuleAttribute);
 
-            Assert.True(descriptor.Rule is TestConsensusRuleWithValidationAttribute);
-            Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithValidationAttribute).RuleName);
-
-            Assert.Equal(3, descriptor.Attributes.Count);
-            Assert.True(descriptor.Attributes[0] is MempoolRuleAttribute);
-            Assert.True(descriptor.Attributes[1] is ExecutionRuleAttribute);
-            Assert.True(descriptor.Attributes[2] is ValidationRuleAttribute);
-
-            Assert.False(descriptor.CanSkipValidation);
+            descriptor.CanSkipValidation.Should().BeFalse();
         }
 
         [Fact]
-        public void Constructor_ConsensusRuleWithOptionalValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithOptionalValidationAttribute_CanSkipValidation()
         {
-            var rule = new TestConsensusRuleWithOptionalValidationAttribute("myRule");
-
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithOptionalValidationAttribute());
 
-            Assert.True(descriptor.Rule is TestConsensusRuleWithOptionalValidationAttribute);
-            Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithOptionalValidationAttribute).RuleName);
+            descriptor.Rule.Should().BeOfType<RuleWithOptionalValidationAttribute>();
+            descriptor.RuleAttributes.Should().HaveCount(1);
+            descriptor.RuleAttributes.Single().Should().BeOfType<ValidationRuleAttribute>();
 
-            Assert.Equal(3, descriptor.Attributes.Count);
-            Assert.True(descriptor.Attributes[0] is MempoolRuleAttribute);
-            Assert.True(descriptor.Attributes[1] is ExecutionRuleAttribute);
-            Assert.True(descriptor.Attributes[2] is ValidationRuleAttribute);
-
-            Assert.True(descriptor.CanSkipValidation);
+            descriptor.CanSkipValidation.Should().BeTrue();
         }
 
         [Fact]
-        public void Constructor_ConsensusRuleWithoutValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithoutValidationAttributeButWithOtherRuleAttribute_CannotSkipValidation()
         {
-            var rule = new TestConsensusRuleWithoutValidationAttribute("myRule");
-
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithMempoolAttributeButWithoutValidationAttribute());
```

Comment:
This rule seems different from the original TestConsensusRuleWithoutValidationAttribute which has 2 attributes. Is that intentional?

Reply:
Yes - the test was only concerned with the validation attribute so the other two were superfluous to the test